pkgs/games/rott/default.nix

```diff
@@ -0,0 +1,61 @@
+{stdenv, lib, fetchurl, SDL, SDL_mixer, makeDesktopItem, copyDesktopItems, runtimeShell, buildShareware ? false}:
+
+let
+  desktopItem = makeDesktopItem {
+    name = "rott";
+    exec = "launch-rott";
+    desktopName = "Rise of the Triad: ${if buildShareware then "The HUNT Begins" else "Dark War"}";
+    categories = "Game;";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "rott";
+  version = "1.1.2";
+
+  src = fetchurl {
+    url = "https://icculus.org/rott/releases/${pname}-${version}.tar.gz";
+    sha256 = "1zr7v5dv2iqx40gzxbg8mhac7fxz3kqf28y6ysxv1xhjqgl1c98h";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems ];
+  buildInputs = [ SDL SDL_mixer ];
+
+  preBuild = ''
+    cd rott
+    make clean
+    make SHAREWARE=${if buildShareware then "1" else "0"}
+  '';
+
+  # Install the executable. In addition, we also create a wrapper script:
+  # launch-rott that changes the current working directory to the data folder:
+  # $HOME/.rott/data
+  #
+  # By default, the executable expects the game data to reside in the same
+  # working directory as the executable making it inconvenient to launch the
+  # program from the user's PATH.
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp rott $out/bin
+
+    cat > $out/bin/launch-rott <<EOF
+    #! ${runtimeShell} -e
+    cd ~/.rott/data
+    exec $out/bin/rott
+    EOF
+
+    chmod +x $out/bin/launch-rott
+
+    runHook postInstall
+  '';
+
+  desktopItems = [ desktopItem ];
```

Comment:
```suggestion
  desktopItems = [
    (makeDesktopItem {
      name = "rott";
      exec = "launch-rott";
      desktopName = "Rise of the Triad: ${if buildShareware then "The HUNT Begins" else "Dark War"}";
      categories = "Game;";
    })
  ];
```

Reply:
This is also a strategy I've seen in some other expressions, but I also have a different preference. I typically give the preference to reduce the amount of nested expressions and rather use in-between values and keep the expression "more flat"

Again, if this is a convention we agree with in the community I'll happily change it.

Reply:
We only move things to let in to prevent them being imported into envs. That is not the case here.

Reply:
I'm not sure if this is always a good general rule of thumb. I can also imagine that for certain packages, there's more additional configuration required, such as generating configuration files and sub packages.

In such cases it's IMO more readable to define these "sub parts" in a let block. Then as a reader you can see that there's a package that itself consists of other parts, without being overwhelmed with the implementation of these sub parts.

Anyway, this kind of practice is obviously a matter of taste.

Since this package only refers to one desktop configuration file, I don't really mind changing it -- the construction of a single configuration file nested inside a derivation should still be readable.